poulpe-service/src/main/java/org/jtalks/poulpe/service/transactional/TransactionalUserService.java

```diff
@@ -247,49 +248,33 @@ public void updateUsersAtGroup(List<User> users, Group group) {
      * {@inheritDoc}
      */
     @Override
-    public PoulpeUser authenticate(String username, String password)
-            throws NotFoundException {
-        PoulpeUser user = getPoulpeUser(username);
-        if (user.getPassword().equals(password)) {
-            return user;
-        } else {
-            throw new NotFoundException();
+    public PoulpeUser authenticate(String username, String password) throws NotFoundException {
+        if (StringUtils.isAnyBlank(username, password)) throw new NotFoundException();
+        List<PoulpeUser> users = requireNotNullNorEmpty(userDao.findUsersByUsernameAndPasswordHash(username, password));
+        switch (users.size()){
+            case 1:
+                return users.get(0);
+            default:
+                return searchForUserCaseSensitive(users, username);
         }
     }
 
+    private PoulpeUser searchForUserCaseSensitive(List<PoulpeUser> users, String username) throws NotFoundException {
+        for (PoulpeUser user : users) {
+            if (user.getUsername().equals(username)) return user;
+        }
+        throw new NotFoundException();
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     public void activate(String uuid) throws NotFoundException, ValidationException {
-        if (uuid == null || uuid.isEmpty()) {
-            throw new NotFoundException();
-        }
-        PoulpeUser user = userDao.getByUUID(uuid);
-        if (user == null) {
-            throw new NotFoundException();
-        } else if (!user.isEnabled()) {
-            user.setEnabled(true);
-            userDao.save(user);
-        } else {
-            throw new ValidationException(Collections.singletonList("user.already_active"));
-        }
-    }
-
-    private PoulpeUser getPoulpeUser(String username) throws NotFoundException {
-        UserSearchRequest searchRequest = new UserSearchRequest(true, Pages.NONE, "username", username);
-        searchRequest.setCaseSensitise(true);
-        List<PoulpeUser> users = userDao.findPoulpeUsersBySearchRequest(searchRequest);
-        if (users.size() == 1) {
-            return users.get(0);
-        } else {
-            for (PoulpeUser user : users) {
-                if (user.getUsername().equals(username)) {
-                    return user;
-                }
-            }
-        }
-        throw new NotFoundException();
+        PoulpeUser user = requireNotNullNorEmpty(userDao.getByUUID(requireNotNullNorEmpty(uuid)));
```

Comment:
Can also be done with Commons Lang:

`PoulpeUser user = Validate.notNull(userDao.getByUUID(requireNotNullNorEmpty(uuid))`